### ISSUE TITLE
[Bug] Cannot turn on flashlight ('Camera "0" has no flashlight' error)

### ISSUE THREAD

**Issue by holecekp** [xamarin/Essentials#1485]
### Description

I have  an application that uses AppCenter for crash reporting. The application calls `Flashlight.TurnOnAsync()` to turn on flashlight. I have received the following error report:

**Java.Lang.IllegalArgumentException: setTorchMode:1791: Camera "0" has no flashlight**

**Stack trace:**
![flashlight error 2](https://user-images.githubusercontent.com/24872113/97625286-9a9e1f00-1a28-11eb-8940-b1d72af911ac.png)

**Device:** P20 lite
**OS:** Android 9

This would be all right for a phone without a flashlight, but P20 lite should have one (see for example: https://www.gsmarena.com/huawei_p20_lite-9098.php).

### Expected Behavior
Flashlight can be turned on, or off on all devices that have one.

### Actual Behavior
Turning flashlight on throws exception on P20 lite.

### Basic Information

- Version with issue: 1.5.3.2
- Last known good version: ?
- IDE: VS 2019
- Platform Target Frameworks: 
  - Android: 10
- Affected Devices: P20 lite

**Comment by jamesmontemagno** [comment]
Is it only on this device? You could grab this code and put it into your app -> https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Flashlight/Flashlight.android.cs#L47 however we check if there is a flashlight so the device is telling us somethign wrong.

I think we can try/catch this -> https://github.com/xamarin/Essentials/blob/main/Xamarin.Essentials/Flashlight/Flashlight.android.cs#L61 and ignore the exception

**Comment by holecekp** [comment]
@jamesmontemagno Unfortunatelly, I don't have the device so I cannot try the code. I have only reports from AppCenter. This specific exception is reported for P20 lite and P10 lite.

Try/Catch is a good idea and I added it also to my code because I have collected other exceptions from AppCenter:
- Java.Lang.Exception: setTorchMode:1774: **Torch for camera "0" is not available due to an existing camera user** (HUAWEI ART-L29; Android 10)
- Java.Lang.IllegalArgumentException: **Bad argument passed to camera service** (Redmi Note 4, Android 6.0)
